Zeeguu Reader/Zeeguu Reader/ZGTextView.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  ZGTextView.swift
+//  Zeeguu Reader
+//
+//  Created by Jorrit Oosterhof on 08-12-15.
+//  Copyright © 2015 Jorrit Oosterhof.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import ZeeguuAPI
+
+class ZGTextView: UITextView {
+
+	var article: Article!
+	
+	convenience init(article: Article) {
+		self.init()
+		self.article = article;
+		self.translatesAutoresizingMaskIntoConstraints = false
+	}
+	
+	func selectedText() -> String {
+//		return self.stringByEvaluatingJavaScriptFromString("window.getSelection().toString()")!
+		return self.textInRange(self.selectedTextRange!)!
+	}
+	
+	func selectedTextContext() -> String {
```

Comment:
this looks cool.
I could think of a few interesting test cases for you if you want :) but then you'd have to make this function testable... which means, take selected range and text as arguments.